content/data/tanzu-gemfire/blog/apache-geode-apache-kafka.md

```diff
@@ -0,0 +1,163 @@
+date: 2020-03-03
+type: blog
+
+# Author(s)
+team: 
+- Jason Huynh
+
+---
+
+## Introduction
+Apache Kafka is a popular solution for ingesting from various data sources and into Kafka topics. For use cases that require key look-ups or querying, traversing the topic for specific keys or devising a strategy for partitioning can be done. However, Kafka also provides the capability to sink, or offload, the data to other systems via Kafka Connect. The [geode-kafka-connector](https://github.com/apache/geode-kafka-connector) implements the Kafka Connect APIs and allows data to be pushed into Apache Geode, where key look-ups and querying are better suited.
+
+[Apache Geode](https://geode.apache.org/) is a consistent, low latency in memory data store. For use cases where we want fast, consistent key look-ups, sinking data to Apache Geode makes a lot of sense. Apache Geode also allows secondary indexes to be built on the data, to be used by queries. The geode-kafka-connector provides a [JsonPdxConverter](https://github.com/apache/geode-kafka-connector/blob/d6651f1ed78c09a533f478ded239a52cd2ffaca3/src/main/java/org/apache/geode/kafka/converter/JsonPdxConverter.java#L27) that we will be using. This allows JSON objects to be converted into PDX ([Portable Data eXchange, an Apache Geode serialization format](https://geode.apache.org/docs/guide/111/developing/data_serialization/gemfire_pdx_serialization.html)), so that we can query any field and drill down into nested JSON Objects.
+
+
+## Overview
+We’ll be ingesting JSON data from a rest endpoint into a Kafka topic. We will sink the data into an Apache Geode region using the geode-kafka-connector and execute queries against the data. Secondary indexes will also be created to allow faster and more efficient query lookups.
+![img](/images/data-blogs/tanzu-gemfire/apache-geode-apache-kafka/diagrams/geode-kafka.png)
+
+(1) Ingest JSON, (2) Connector pulls JSON data from topic, (3) geode-kafka-connector converts to PDX and pushes to Apache Geode, (4) Query is executed in GFSH, (5) Results are returned
+
+
+## Prerequisites:
+* [Apache Geode installed](https://geode.apache.org/releases/) or use the [docker image](https://hub.docker.com/r/apachegeode/geode/)
+* [Apache Kafka installed](https://kafka.apache.org/downloads)
+* Download or build a [geode-kafka-connector jar](https://github.com/apache/geode-kafka-connector)
+* Download the source for an example ingest web service: [https://github.com/jhuynh1/spring-kafka-JSON-ingest](https://github.com/jhuynh1/spring-kafka-JSON-ingest)
+
+
+## Start Apache Geode Cluster
+Execute gfsh from your Apache Geode Installation and start a locator and a server:
+```
+./gfsh 
+ start locator --name=locator1
+ start server --name=server1
+```
+
+![img](/images/data-blogs/tanzu-gemfire/apache-geode-apache-kafka/geode-kafka-gfsh.png)
+
+gfsh output after creating locator and server
+
+
+Create the region we want the data to end up in:
+create region --name=’Events’ --type=PARTITION
```

Comment:
suggest wrapping in a code block if this is supposed to be a command the reader is intended to copy and paste out